indexer/editable_map_object.cpp

```diff
@@ -578,6 +579,217 @@ bool EditableMapObject::ValidateName(string const & name)
+      break;
+    }
+    case JournalEntryType::LegacyObject:
+    {
+      ASSERT_FAIL(("Legacy Objects can not be loaded from Journal"));
+      break;
+    }
+  }
+}
+
+void EditableMapObject::LogDiffInJournal(EditableMapObject const & unedited_emo)
+{
+  LOG(LDEBUG, ("Executing LogDiffInJournal"));
+
+  // Name
+  for (StringUtf8Multilang::Lang language : StringUtf8Multilang::GetSupportedLanguages())
+  {
+    int8_t langCode = StringUtf8Multilang::GetLangIndex(language.m_code);
+    std::string_view new_name;
+    std::string_view old_name;
+    m_name.GetString(langCode, new_name);
+    unedited_emo.GetNameMultilang().GetString(langCode, old_name);
+
+    if (new_name != old_name)
+    {
+      std::string osmLangName = StringUtf8Multilang::GetOSMTagByCode(langCode);
+      m_journal.AddTagChange(osmLangName, std::string(old_name), std::string(new_name));
+    }
+  }
+
+  // Address
+  if (m_street.m_defaultName != unedited_emo.GetStreet().m_defaultName)
+    m_journal.AddTagChange("addr:street", unedited_emo.GetStreet().m_defaultName, m_street.m_defaultName);
+
+  if (m_houseNumber != unedited_emo.GetHouseNumber())
+    m_journal.AddTagChange("addr:housenumber", unedited_emo.GetHouseNumber(), m_houseNumber);
+
+  // Metadata
+  for (uint8_t i = 0; i < static_cast<uint8_t>(feature::Metadata::FMD_COUNT); ++i)
+  {
+    auto const type = static_cast<feature::Metadata::EType>(i);
+    std::string const & value = std::string(GetMetadata(type));
+    std::string const & old_value = std::string(unedited_emo.GetMetadata(type));
+
+    if (value != old_value)
+      m_journal.AddTagChange(ToString(type), old_value, value);
+  }
+
+  // cuisine and diet
+  std::vector<std::string> new_cuisines = GetCuisines();
+  std::vector<std::string> old_cuisines = unedited_emo.GetCuisines();
+
+  auto const findAndErase = [] (std::vector<std::string>* cuisinesPtr, std::string s)
+  {
+    auto it = std::find((*cuisinesPtr).begin(), (*cuisinesPtr).end(), s);
+    if (it != (*cuisinesPtr).end())
+    {
+      (*cuisinesPtr).erase(it);
+      return "yes";
+    }
+    return "";
+  };
+
+  std::string new_vegetarian = findAndErase(&new_cuisines, "vegetarian");
+  std::string old_vegetarian = findAndErase(&old_cuisines, "vegetarian");
+  if (new_vegetarian != old_vegetarian)
+    m_journal.AddTagChange("diet:vegetarian", old_vegetarian, new_vegetarian);
+
+  std::string new_vegan = findAndErase(&new_cuisines, "vegan");
+  std::string old_vegan = findAndErase(&old_cuisines, "vegan");
+  if (new_vegan != old_vegan)
+    m_journal.AddTagChange("diet:vegan", old_vegan, new_vegan);
+
+  bool cuisinesModified = false;
+
+  if (new_cuisines.size() != old_cuisines.size())
+    cuisinesModified = true;
+  else
+  {
+    for (auto const & new_cuisine : new_cuisines)
+    {
+      for (auto const & old_cuisine : old_cuisines)
+      {
+        if (new_cuisine == old_cuisine)
+          goto value_found;
+      }
+      cuisinesModified = true;
+      value_found:;
```

Comment:
```
if (!base::IsExist(old_cuisines, new_cuisine))
{
  cuisinesModified = true;
  break;
}
```

Reply:
Thanks! That makes the code way cleaner than with the `goto`